### ISSUE TITLE
Update LWRP Options for Filebeat 5.2

### ISSUE THREAD

**Issue by jamiebuxxx** [vkhatri/chef-filebeat#100]
Hello, 

We recently upgrade our Filebeat version to 5.2 and everything appears to be working fine. Today when we tried to update the prospector configuration option for [close_timeout](https://www.elastic.co/guide/en/beats/filebeat/current/configuration-filebeat-options.html#close-timeout), I noticed there was not LWRP option for that. Instead there is `close_older` which is now deprecated in version 5.2. 

Instead of just creating a pull request to the changes I need, I want to open the discussion to get opinions on methods to include the latest Filebeat options with previous options.

**Comment by kylekrall** [comment]
👍 

And for those that are having SSL issues with 5.2.2 "ERR Failed to publish events caused by: EOF" the easiest fix is to change change the `tls` attribute to `ssl` (in a wrapper cookbook), as TLS was renamed to SSL in Filebeat 5.0.

https://discuss.elastic.co/t/filebeat-err-failed-to-publish-events-caused-by-eof/65105

**Comment by vkhatri** [comment]
@jamiebuxxx Will add recent `filebeat` versions configuration parameters to LWRP.

Perhaps we can filter options for various `filebeat` versions.

https://github.com/vkhatri/chef-filebeat/blob/master/libraries/provider_prospector.rb#L37

Will check it out. Sorry for the extreme delay in response.

**Comment by jamiebuxxx** [comment]
@vkhatri No worries. Thanks for checking this out. I think filter options per versions would be a good idea. And creates a template for future version upgrade.

**Comment by vkhatri** [comment]
Looks like there are three categories of options to maintain in LWRP.

```shell
- options for all filebeat versions
- options added by a filebeat version
- options deprecated by a filebeat version
```

This is currently applicable to prospectors added using LWRP. Prospectors added using node attribute `node['filebeat']['prospectors']` does not use LWRP, may be it should.

**Comment by jamiebuxxx** [comment]
Thats a good point about adding prospectors via the node attribute. It should the the LWRP whenever possible. Even if it just loop of a hash of entries in the node attribute.

**Comment by vkhatri** [comment]
Maintaining filebeat version specific attributes will require a lot of effort and continuous maintenance. I found it is easier to leave it up to the user to choose the right LWRP options. 

However, new attributes have been added to the LWRP.

The cookbook release `v1.0.0` will no longer actively support/test filebeat `1.x`.

Also, prospectors via node attribute now using LWRP.

Filebeat configuration documentation is kept minimal as possible and users are encouraged to reference filebeat documentation instead. IMO, `attributes/config.rb` provides sufficient reference to get started.

Fixed in master. Keeping this issue open for some more time.
